tests/integration/kubernetes/tests_common.sh

```diff
@@ -359,3 +359,14 @@ pod_exec_blocked_command() {
 
 	(echo "${exec_output}" | grep "ExecProcessRequest is blocked by policy" > /dev/null) || die "exec was not blocked by policy!"
 }
+
+# Print the node journal since the test start time if a bats test is not completed
+output_journal_if_test_incomplete() {
```

Comment:
I think this is fine at the moment. I wonder if there is value in mirroring  that all these tests having `setup_common` calls in their `setup` method, but introducing `teardown_common` here and calling `output_journal_if_test_incomplete "${node}" "${node_start_time:-}"` from it and then calling `teardown_common` in the tests `teardown` method?

It seems like a few of these tests are also calling:
```
    kubectl describe pods
    k8s_delete_all_pods_if_any_exists || true
```
or variants of this:
```
	[ -n "${pod_name:-}" ] && kubectl describe "pod/${pod_name}" || true
	[ -n "${pod_config_dir:-}" ] && kubectl delete -f "${K8S_TEST_YAML}" || true
```
for debug and cleanup purposes, so moving them into `teardown_common` too might make sense?

No mandatory, but might help reduce even more duplication?

Reply:
It sounds like a great plan. I will refactor the code. Thanks for feedback. 😉